In [1]:
# Importing necessary libraries
import numpy as np
import pandas as pd
import scipy.stats as stats #for statistical functions / operations

In [2]:
# given Sample data for the durability of print-heads (in millions of characters)
durability_data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 
                            1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])
# Creating a DataFrame for ease of working
df = pd.DataFrame(durability_data, columns=['Durability'])

### Step 1: Calculating sample mean and sample standard deviation

In [4]:
n = len(df)
sample_mean = df['Durability'].mean()
sample_std = df['Durability'].std(ddof=1)  # Sample standard deviation , ddof=1  corrects the bias in the estimation of the population standard deviation from a sample
print(f"Sample Mean: {sample_mean:.3f}")
print(f"Sample Standard Deviation: {sample_std:.3f}")

Sample Mean: 1.239
Sample Standard Deviation: 0.193


### Step 2: Calculating the t-value for 99% confidence level

In [6]:
confidence_level = 0.99
alpha = 1 - confidence_level
t_value = stats.t.ppf(1 - alpha/2, n - 1)# t.ppf=(percent-point function) is the inverse of the cumulative distribution function (CDF) for the t-distribution. Essentially, it returns the t-value associated with a given probability level.
t_value

2.976842734370834

### Step 3: Constructing the confidence interval using Sample Standard deviation

In [8]:
margin_of_error_sample = t_value * (sample_std / np.sqrt(n))
ci_sample = (sample_mean - margin_of_error_sample, sample_mean + margin_of_error_sample)
print(f"99% Confidence Interval (Sample Std Dev): {ci_sample}")
# Part B: Using known population standard deviation
population_std = 0.2  # Known population standard deviation

99% Confidence Interval (Sample Std Dev): (1.0901973384384906, 1.3871359948948425)


### Step 4: Calculating the z-value for 99% confidence level

In [10]:
z_value = stats.norm.ppf(1 - alpha/2)

### Step 5: Constructing the confidence interval using known Population standard deviation

In [12]:
margin_of_error_population = z_value * (population_std / np.sqrt(n))
ci_population = (sample_mean - margin_of_error_population, sample_mean + margin_of_error_population)
print(f"99% Confidence Interval (Known Pop Std Dev): {ci_population}")


99% Confidence Interval (Known Pop Std Dev): (1.1056514133957607, 1.3716819199375725)


### conclusion 

Both confidence intervals provide valuable insights into the durability of the print-heads, with both intervals indicating that the mean durability is expected to be just over 1 million characters but less than 1.4 million characters. The slight differences between the intervals highlight the impact of knowing the population standard deviation versus relying on the sample standard deviation, emphasizing the importance of understanding the underlying assumptions in statistical analysis.

These results can guide the manufacturer in assessing product quality and making informed decisions regarding production and quality control processes.